### ISSUE TITLE
Poetry fails to install multiple dependencies form the same git repo in parallel

### ISSUE THREAD

**Issue by giner** [python-poetry/poetry#7949]
- **Poetry version**: 1.4.2
- **Python version**: 3.11.2
- **OS version and name**: Ubuntu 22.04
- **pyproject.toml**: https://gist.github.com/giner/c84029441815f0b54da31b546a3b2c7f

- [x] I am on the [latest](https://github.com/python-poetry/poetry/releases/latest) stable Poetry version, installed using a recommended method.
- [x] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate.
- [x] I have consulted the [FAQ](https://python-poetry.org/docs/faq/) and [blog](https://python-poetry.org/blog/) for any relevant entries or release notes.
- [x] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option) and have included the output below.

## Issue

Poetry fails to install multiple modules form the same git repo in parallel with random errors such as the following:
```
poetry install
...
  [Errno 2] No such file or directory: b'/home/user/.cache/pypoetry/virtualenvs/wallet-T3t3_RZf-py3.11/src/poetry-modules/.git/HEAD.lock' -> b'/home/user/.cache/pypoetry/virtualenvs/wallet-T3t3_RZf-py3.11/src/poetry-modules/.git/HEAD'
...
```

Workaround I use at the moment is disabling `parallel` for the installer:
```
poetry config installer.parallel false
```

**Comment by ralbertazzi** [comment]
Duplicate of https://github.com/python-poetry/poetry/issues/6958

**Comment by giner** [comment]
Looks similar however `1.2.2` had an old installer

**Comment by ogreyesp** [comment]
Hi all

This is still happening. Poetry doesn´t give any error, but when I check in the python environment, one of the dependencies that comes from the same repo (but from different branch) was not installed.

**Comment by dimbleby** [comment]
you don't need to comment on open issues only to say that they should still be open.  The open issue already tracks that.

having said which this one has already been identified as a duplicate of #6958 and should be closed in favour of that

**Comment by ogreyesp** [comment]
@dimbleby 

My comment aims at seeing if someone has found a solution for this, at least a temporal one... Considering that Poetry is being used nowadays in productions environments in several private companies, I think that this error is important and it seems that there is not movement on this bug since May, 19th. I´m right now struggling in my company with this bug. 

Anyway, if you are upset for my comment, maybe you don´t understand what is an open source project, a pity....

**Comment by dimbleby** [comment]
> My comment aims at seeing if someone has found a solution for this, at least a temporal one.

you didn't write that, though, you wrote "this is still happening" - which everyone knows, because the issue is still open.

useful conversation is welcome and encouraged.

(but this one is still a duplicate and ought to be closed!  if there is useful conversation to be had it would be better to have it all in #6958 and not split that into two)

**Comment by radoering** [comment]
Closing as duplicate of #6958
